cfn-resources/api-key/cmd/resource/resource.go

```diff
@@ -0,0 +1,527 @@
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/secrets"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+var CreateRequiredFields = []string{constants.OrgID, constants.Description}
+var UpdateRequiredFields = []string{constants.OrgID, constants.APIUserID, constants.Description}
+var ReadRequiredFields = []string{constants.OrgID, constants.APIUserID}
+var DeleteRequiredFields = []string{constants.OrgID, constants.APIUserID}
+var ListRequiredFields = []string{constants.OrgID}
+
+type APIKeySecret struct {
+	APIUserID  string
+	PublicKey  string
+	PrivateKey string
+}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-api-key")
+}
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(CreateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlas, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	// Set the roles from model
+	apiKeyInput := atlasSDK.CreateAtlasOrganizationApiKey{
+		Desc:  currentModel.Description,
+		Roles: currentModel.Roles,
+	}
+	apiKeyRequest := atlas.AtlasV2.ProgrammaticAPIKeysApi.CreateApiKey(
+		context.Background(),
+		*currentModel.OrgId,
+		&apiKeyInput,
+	)
+	apiKeyUserDetails, response, err := apiKeyRequest.Execute()
+	defer closeResponse(response)
+	if err != nil {
+		return handleError(response, constants.CREATE, err)
+	}
+
+	// Read response
+	currentModel.APIUserId = apiKeyUserDetails.Id
+
+	// Save PrivateKey in AWS SecretManager
+	secret := APIKeySecret{APIUserID: *currentModel.APIUserId, PublicKey: *apiKeyUserDetails.PublicKey, PrivateKey: *apiKeyUserDetails.PrivateKey}
+	secretName := *currentModel.APIUserId
+	if currentModel.AwsSecretName != nil {
+		secretName = *currentModel.AwsSecretName
+	}
+	_, err = secrets.Create(&req, secretName, secret, currentModel.Description)
```

Comment:
can we check if the secret exists, if it exists we can update it? probably the user created the secret in the stack and wats the api key to be store there,